pkg/cloud/ovirt/providerIDcontroller/providerIDController.go

```diff
@@ -0,0 +1,150 @@
+package providerIDcontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-api-provider-ovirt/pkg/cloud/ovirt/clients"
+)
+
+var _ reconcile.Reconciler = &providerIDReconciler{}
+
+type providerIDReconciler struct {
+	log                  logr.Logger
+	client               client.Client
+	listNodesByFieldFunc func(key, value string) ([]corev1.Node, error)
+	fetchProviderIDFunc  func(string) (string, error)
+	ovirtApi             *ovirtsdk.Connection
+}
+
+func (r *providerIDReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.log.WithValues("node", request.NamespacedName)
+	r.log.V(3).Info("Reconciling", request.NamespacedName)
```

Comment:
I don't think you're using the `log.WithValues` right here, I think you have to assign the result of `WithValues` to something for it to have any effect, eg

```
log := r.log.WithValues("node", request.NamespacedName)
log.V(3).Info("Reconciling node")
```

Which should then include the `node: <name>` key value pair on all log lines produced by `log`

Reply:
oh I thought this would mutate the current log instance